ptf

```diff
@@ -133,6 +134,14 @@ be subtracted from the result by prefixing them with the '^' character.  """
             # Parse --interface
             def check_interface(value):
                 try:
+                    sp = ';'
+                    if sp in value:
+                        value, p_info = value.split(sp, 1)
+                        params = p_info.split(sp)
+                        port = params.pop(0)
+                        params.pop()
+                        port_cfg = dict(val.split("=") for val in params)
+                        getattr(namespace, "port_info")[port] = port_cfg
```

Comment:
why do we introduce a new `"port_info"` dictionary instead of adding this information to the interfaces list?

Reply:
Today the interfaces list is a list of device id, port id, interface name tuples.  If we add a dictionary with these per-port key-value pairs it may cause issues with existing script using PTF and accessing config["interfaces"]; instead of unpacking 3 values it is now 4.
Also, this makes it similar to the test-params dictionary and easy to implement the get function in testutils.py.  It can lookup the dictionary using the port-id instead of walking the interface list looking for the port.

Reply:
Thanks for clarifying Steve!